src/zenml/integrations/gcp/step_operators/vertex_step_operator.py

```diff
@@ -265,6 +265,7 @@ def launch(
             }
             if self.config.encryption_spec_key_name
             else {},
+            "persistent_resource_id": settings.persistent_resource_id,
```

Comment:
@htahir1 Did you actually test this? From what I can see on the docs, this field does not exist on the custom job object that expected to be passed to create a custom job: https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1.types.CustomJob

Instead, it seems to be available in the spec of that custom job (https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1beta1.types.CustomJobSpec)